planet/cli/data.py

```diff
@@ -23,83 +45,66 @@ def parse_item_types(ctx, param, value: str) -> List[str]:
-# This decorator gives our function a context (named "ctx") that may
-# contain parameters set by the "planet data" command group, like
-# "planet --quiet data" or "planet --verbosity=debug data".
-@click.pass_context
-async def search_quick(ctx,
-                       item_types: List[str],
-                       filter: dict,
-                       limit: int,
-                       pretty: bool,
-                       name: str):
-    """Execute a structured item search and print results.
+              help='Maximum number of results to return. Defaults to 100.')
+async def search_quick(ctx, item_types, filter, name, limit, pretty):
+    """Execute a structured item search.
+    Quick searches are stored for approximately 30 days and the --name
+    parameter will be applied to the stored quick search. \n
 
-    Results are represented as a sequence of GeoJSON features.
+    Arguments: \n
+    ITEM_TYPES - string. Comma-separated item type identifier(s). \n
+    FILTER - string. A full JSON description of search criteria.
+    Supports file and stdin. \n
 
-    """
+    Output:
+    A series of GeoJSON descriptions for each of the returned items.
 
-    # Note: API tokens will be found in "ctx". They are unused in this
-    # example.
-    #
-    # The Planet Data client will return a stream of GeoJSON
-    # Feature-like dicts (aka an "iterator"). The search_quick function
-    # will iterate over the features, encode them as JSON, and echo
-    # them. We simulate that here.
-    async def example_client(item_types,
-                             filter,
-                             limit=0,
-                             name=None) -> AsyncIterator[dict]:
-        """This is a placeholder.
-
-        We'll delete this and use the real method from
-        planet.clients.data when it is available.
-
-        """
-        # Note: we re-emit the input item types and filter to help us
-        # sanity check during early sprints.
-        for feature in item_types:
-            yield dict(item_type=feature)
-
-        yield dict(filter=filter)
-
-    # CLI functions raise ClickException when any kind of intentionally
-    # raised PlanetError occurs. We *don't* handle other errors now
-    # because those will be caused by bugs in our code and we want them
-    # raw and unfiltered.
-    async for feature in example_client(item_types,
-                                        filter,
-                                        limit=limit,
-                                        name=name):
-        echo_json(feature, pretty=pretty)
+    """
+    async with data_client(ctx) as cl:
+        items = await cl.quick_search(name=name,
+                                      item_types=item_types,
+                                      search_filter=filter,
+                                      limit=limit,
+                                      sort=None)
+        async for item in items:
```

Comment:
@jreiberkyle remember our discussion about having quick_search fail fast when there are no result items? I had a related thought while working on the above with Mansi.

My first take at iterating over the items was this:

```python
    async with data_client(ctx) as cl:
        items = await cl.quick_search(name=name,
                                      item_types=item_types,
                                      search_filter=filter,
                                      limit=limit,
                                      sort=None)
    async for item in items:
        echo_json(item, pretty)
```

I tried to iterate outside of the data_client block and that did not work and had a not obvious error. I suspect that users might stumble on this too. The `items` object looks like it is independent of the client, but it is not (on account of the do_request_fn).

What would you think about changes to allow the fool-proof usage below?

```python
async with data_client(ctx) as cl:
    async for item in cl.quick_search(...):
        echo_json(item)
```